### DISCUSSION TITLE
S3Error : S3 operation failed; code: IncompleteBody

### DISCUSSION THREAD

**Discussion by manoranjan1101** [minio/minio#19786] [Q&A]
I am using minio with kubeflow in a microk8s cluster. A minio client (v7.2.7) in python connects to the bucket to fetch and upload data. Occasionally I am encountering this error: 
_S3Error : S3 operation failed; code: IncompleteBody, message: You did not provide the number of bytes specified by the Content-Length HTTP header_

This error is not always popping up and goes away after sometime or restarting the cluster. Any idea why this error is coming?

**Comment by balamurugana** [comment]
Please check your server/proxy/load balancer etc.  `IncompleteBody` means that the server didn't receive expected bytes.

**Comment by manoranjan1101** [comment]
Thanks for your quick suggestion. I am trying to understand how this error is not popping regularly and occurs over random intervals. Is it a problem within the kubernetes cluster?

**Comment by balamurugana** [comment]
Yep. It is mostly your Kubernetes issue. If you are using any proxy/load balancer, check that as well.

**Comment by manoranjan1101** [comment]
Thanks, will investigate it further
